src/network.cpp

```diff
@@ -193,6 +195,33 @@ address Network::fetchSubnetMask(const std::string& interface)
         "Interface {} does not have a netmask address defined", interface));
 }
 
+void Network::calculateAddress(const address& connectionAddress)
```

Comment:
I think this method should return instead of setting directly. That's for consistency. If you look at `fetchAddress` it will fetch and return, and them was the setter job to set the address on the object.

Reply:
Done. https://github.com/viniciusferrao/cloysterhpc/pull/23/commits/878fd75d228a94ea2c80f6ec104806662e5e303b